website/addons/googledrive/tests/test_views.py

```diff
@@ -5,183 +5,80 @@
         self.project = ProjectFactory(creator=self.user)
         self.project.add_addon('googledrive', Auth(self.user))
         self.node_settings = self.project.get_addon('googledrive')
         self.node_settings.user_settings = self.user_settings
-        oauth_settings = GoogleDriveOAuthSettingsFactory()
-        oauth_settings.save()
-        self.user_settings.oauth_settings = oauth_settings
         self.node_settings.save()
+        self.user_settings.save()
         # Log user in
         self.app.authenticate(*self.user.auth)
 
-    def test_drive_user_config_get_returns_correct_urls(self):
-        self.user.add_addon('googledrive')
-        url = api_url_for('googledrive_user_config_get', Auth(self.user))
+    def test_list_googledrive_accounts_returns_accounts_single(self):
+        url = api_url_for('list_googledrive_user_accounts', Auth(self.user))
         res = self.app.get(url)
         assert_equal(res.status_code, 200)
-        result = res.json['result']['urls']
-        assert_true(result['create'], api_url_for('googledrive_oauth_start_user'))
-        assert_true(result['delete'], api_url_for('googledrive_oauth_delete_user'))
+        assert_equal(len(res.json['accounts']), 1)
+        assert_equal(res.json['accounts'][0]['provider_id'], self.account.provider_id)
+        assert_equal(res.json['accounts'][0]['provider_name'], self.account.provider_name)
 
-    def test_drive_user_config_get_has_auth(self):
-        self.user.add_addon('googledrive')
-        user_settings = self.user.get_addon('googledrive')
-        user_settings.access_token = 'abc123'
-        user_settings.save()
-        url = api_url_for('googledrive_user_config_get', Auth(self.user))
+    def test_list_googledrive_accounts_returns_accounts_multiple(self):
+        external_account = GoogleDriveAccountFactory()
+        self.user.external_accounts.append(external_account) # self.account is already present
+        self.user.save()
+        url = api_url_for('list_googledrive_user_accounts', Auth(self.user))
         res = self.app.get(url)
         assert_equal(res.status_code, 200)
-        result = res.json['result']
-        assert_true(result['userHasAuth'])
+        assert_equal(len(res.json['accounts']), 2)
 
-    def test_drive_user_config_get_not_has_auth(self):
-        self.user_settings.access_token = None
-        self.user_settings.save()
-        url = api_url_for('googledrive_user_config_get', Auth(self.user))
+    def test_googledrive_config_get_return_correct_urls(self):
+        # self.node_settings.set_auth(external_account=self.account, user=self.user)
+        url = self.project.api_url_for('googledrive_config_get', Auth(self.user))
+        res = self.app.get(url)
+        result = res.json['result']
+        assert_equal(result['urls']['accounts'],  self.project.api_url_for('list_googledrive_user_accounts'))
+        assert_equal(result['urls']['auth'], api_url_for('oauth_connect',
```

Comment:
I was originally going to annotate the first few api_url_for usages to mark each occurrence. But I think you can handle the full-text search of the diff from here. :)